### ISSUE TITLE
Pyproj error running makeh5file

### ISSUE THREAD

**Issue by danielmwatkins** [WilhelmusLab/ice-floe-tracker-pipeline#51]
There's an issue trying to run the makeh5file, both in cylc and when running in package mode. Looks like it can't find the pyproj library. Here's the error I get:

```
julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p resources/truecolor/20230526.aqua.truecolor.250m.tiff -r results/preprocess

ERROR: LoadError: PyError ($(Expr(:escape, :(ccall(#= /Users/dwatkin2/.julia/packages/PyCall/ilqDX/src/pyeval.jl:38 =# @pysym(:PyEval_EvalCode), PyPtr, (PyPtr, PyPtr, PyPtr), o, globals, locals))))) <class 'ModuleNotFoundError'>
ModuleNotFoundError("No module named 'pyproj'")
```

**Comment by cpaniaguam** [comment]
I just tested this again using the input files in [testresultsdata.zip](https://github.com/WilhelmusLab/ice-floe-tracker-pipeline/files/12316957/testresultsdata.zip), which are also used in the CI tests for the package on GitHub, and it seems to have worked 🤔. Here is the output I got:
```bash
cpaniagu@CIS14Y0NRQ3 MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p results/testdatafiles/20220914.aqua.truecolor.250m.tiff  -r results/testdatafiles
  Activating project at `C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline`
  3.475010 seconds (3.13 M allocations: 223.570 MiB, 1.87% gc time, 87.10% compilation time: 3% of which was recompilation)

cpaniagu@CIS14Y0NRQ3 MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ ls -l results/testdatafiles/
total 1354
-rw-r--r-- 1 cpaniagu 1425514 1273587 Jun 20 11:45 20220914.aqua.truecolor.250m.tiff
-rw-r--r-- 1 cpaniagu 1425514     240 Aug 10 15:51 filenames.jls
-rw-r--r-- 1 cpaniagu 1425514    2226 Aug 10 15:51 floe_props.jls
drwxr-xr-x 1 cpaniagu 1425514       0 Aug 10 16:17 hdf5-files/
-rw-r--r-- 1 cpaniagu 1425514      63 Aug 10 15:51 passtimes.jls
-rw-r--r-- 1 cpaniagu 1425514  106088 Aug 10 15:51 segmented_floes.jls

cpaniagu@CIS14Y0NRQ3 MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ ls -l results/testdatafiles/hdf5-files/
total 19928
-rw-r--r-- 1 cpaniagu 1425514 10201256 Aug 10 16:31 20220914T1244.aqua.labeled_image.250m.h5
-rw-r--r-- 1 cpaniagu 1425514 10201256 Aug 10 16:31 20220914T1359.terra.labeled_image.250m.h5
```
You can inspect the generated h5 files within the `hdf5-files` folder.

Upon closely reading the error message I get using @danielmwatkins data, it seems like the `results-daniel/20230526.aqua.truecolor.250m.tiff` image file doesn't have the coordinates data? The first line in the error message I get is this:
```bash
cpaniagu@CIS14Y0NRQ3 MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p results-daniel/20230526.aqua.truecolor.250m.tiff -r results-daniel/preprocess
  Activating project at `C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline`
C:\Users\cpaniagu\.julia\conda\3\x86_64\lib\site-packages\rasterio\__init__.py:304: NotGeoreferencedWarning: Dataset has
 no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
```
Maybe @ellenbuckley and @mkkim400 could try the pipeline with result files generated using different date ranges? That way we can rule out the possibility of some files not coming with correct georeferencing data. Thanks everyone! :-)

**Comment by danielmwatkins** [comment]
That would make sense for the error message you got when you ran it earlier today, but it doesn’t explain why I was getting an error where it couldn’t find pyproj. I’ll see if I can get the crs for the image via rasterio.-DanielOn Aug 10, 2023, at 16:53, Carlos Paniagua ***@***.***> wrote:﻿
I just tested this again using the input files in testresultsdata.zip, which are also used in the CI tests for the package on GitHub, and it seems to have worked 🤔. Here is the output I got:
***@***.*** MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p results/testdatafiles/20220914.aqua.truecolor.250m.tiff  -r results/testdatafiles
  Activating project at `C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline`
  3.475010 seconds (3.13 M allocations: 223.570 MiB, 1.87% gc time, 87.10% compilation time: 3% of which was recompilation)

***@***.*** MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ ls -l results/testdatafiles/
total 1354
-rw-r--r-- 1 cpaniagu 1425514 1273587 Jun 20 11:45 20220914.aqua.truecolor.250m.tiff
-rw-r--r-- 1 cpaniagu 1425514     240 Aug 10 15:51 filenames.jls
-rw-r--r-- 1 cpaniagu 1425514    2226 Aug 10 15:51 floe_props.jls
drwxr-xr-x 1 cpaniagu 1425514       0 Aug 10 16:17 hdf5-files/
-rw-r--r-- 1 cpaniagu 1425514      63 Aug 10 15:51 passtimes.jls
-rw-r--r-- 1 cpaniagu 1425514  106088 Aug 10 15:51 segmented_floes.jls

***@***.*** MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ ls -l results/testdatafiles/hdf5-files/
total 19928
-rw-r--r-- 1 cpaniagu 1425514 10201256 Aug 10 16:31 20220914T1244.aqua.labeled_image.250m.h5
-rw-r--r-- 1 cpaniagu 1425514 10201256 Aug 10 16:31 20220914T1359.terra.labeled_image.250m.h5
You can inspect the generated h5 files within the hdf5-files folder.
Upon closely reading the error message I get using @danielmwatkins data, it seems like the results-daniel/20230526.aqua.truecolor.250m.tiff image file doesn't have the coordinates data? The first line in the error message I get is this:
***@***.*** MINGW64 ~/Documents/ice-floe-tracker-pipeline (main)
$ julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p results-daniel/20230526.aqua.truecolor.250m.tiff -r results-daniel/preprocess
  Activating project at `C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline`
C:\Users\cpaniagu\.julia\conda\3\x86_64\lib\site-packages\rasterio\__init__.py:304: NotGeoreferencedWarning: Dataset has
 no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
Maybe @ellenbuckley and @mkkim400 could try the pipeline with result files generated using different date ranges? That way we can rule out the possibility of some files not coming with correct georeferencing data. Thanks everyone! :-)

**Comment by cpaniaguam** [comment]
> That would make sense for the error message you got when you ran it earlier today, but it doesn’t explain why I was getting an error where it couldn’t find pyproj.

Indeed. Just trying to narrow down possible problems!

I also tested things on Oscar with the same test files and it also worked. After cloning the repo and placing the test files in the correct directory, I lauched julia, activated/instantiated the project, and ran the command.
```julia
[cpaniagu@node1317 ~]$ module load julia/1.9.2
module: unloading 'julia/1.9.0'
module: loading 'julia/1.9.2'
[cpaniagu@node1317 ~]$ julia -v
julia version 1.9.2
[cpaniagu@node1317 ~]$ cd scratch/ice-floe-tracker-pipeline
[cpaniagu@node1317 ice-floe-tracker-pipeline]$ julia -q
(v1.9) pkg> activate .
  Activating project at `/oscar/scratch/cpaniagu/ice-floe-tracker-pipeline`

(IFTPipeline) pkg> instantiate
Precompiling project...
  117 dependencies successfully precompiled in 550 seconds. 102 already precompiled.

julia> exit()
[cpaniagu@node1317 ice-floe-tracker-pipeline]$ julia --project=. workflow/scripts/ice-floe-tracker.jl makeh5files -p results/test/20220914.aqua.truecolor.250m.tiff -r results/test/
  6.012452 seconds (3.51 M allocations: 248.725 MiB, 2.26% gc time, 85.72% compilation time: 4% of which was recompilation)
[cpaniagu@node1317 ice-floe-tracker-pipeline]$
```

**Comment by danielmwatkins** [comment]
Just an update - the images downloaded from fetchdata.sh definitely still
have the geotiff crs information. I'm able to run latlon.py locally.

**Comment by cpaniaguam** [comment]
@danielmwatkins Thanks for the feedback!

**Comment by cpaniaguam** [comment]
> Just an update - the images downloaded from fetchdata.sh definitely still have the geotiff crs information. I'm able to run latlon.py locally.

Did you try the image in the [zip file above](https://github.com/WilhelmusLab/ice-floe-tracker-pipeline/files/12316957/testresultsdata.zip)? There is something unusual about it. I tried the first steps of `getlatlon` with this image:
```python
>>> import rasterio # v1.3.7
>>> imgpath  = r"C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline\results-daniel\20230526.aqua.truecolor.250m.tiff"
>>> im = rasterio.open(imgpath)
>>> im.crs.__str__()  # used later in Transformer.from_crs
'None'
```

If one later calls `pyproj.Transformer.from_crs` passing `im.crs` with value `None` an error is thrown:
```python
pyproj.exceptions.CRSError: Invalid projection: : (Internal Proj Error: proj_create: unrecognized format / unknown name)
```

Also, at times ` rasterio.open` fails when passing a path to this image. I am not sure if this is a bug in `rasterio`.

```python
$ python
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import rasterio # using v1.3.7
>>> imgpath = r"C:\Users\cpaniagu\Documents\ice-floe-tracker-pipeline\results-daniel\20230526.aqua.truecolor.250m.tiff"
>>> im = rasterio.open(imgpath)
C:\Users\cpaniagu\AppData\Local\Programs\Python\Python311\Lib\site-packages\rasterio\__init__.py:319: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
```
If I try again in the same session it runs -- really strange.

There is a newer version of `rasterio` out. I will try that one to see what happens. Please try processing the suspicious image on your end to see if you get the same result. Thanks!

**Comment by tdivoll** [comment]
Fixed by #63
